src/Symfony/Component/Yaml/Tests/DumperTest.php

```diff
@@ -403,6 +415,59 @@ public function testDumpingTaggedValueSequenceRespectsInlineLevel()
 
 YAML;
         $this->assertSame($expected, $yaml);
+        $this->assertSameData($data, $this->parser->parse($expected, Yaml::PARSE_CUSTOM_TAGS));
+    }
+
+    public function testDumpingTaggedValueTopLevelScalar()
+    {
+        $data = new TaggedValue('user', 'jane');
+
+        $yaml = $this->dumper->dump($data);
+
+        $expected = '!user jane';
+        $this->assertSame($expected, $yaml);
+        $this->assertSameData($data, $this->parser->parse($yaml, Yaml::PARSE_CUSTOM_TAGS));
+    }
+
+    public function testDumpingTaggedValueTopLevelAssocInline()
+    {
+        $data = new TaggedValue('user', ['name' => 'jane']);
+
+        $yaml = $this->dumper->dump($data);
+
+        $expected = '!user { name: jane }';
+        $this->assertSame($expected, $yaml);
+        $this->assertSameData($data, $this->parser->parse($yaml, Yaml::PARSE_CUSTOM_TAGS));
+    }
+
+    public function testDumpingTaggedValueTopLevelAssoc()
+    {
+        $data = new TaggedValue('user', ['name' => 'jane']);
+
+        // @todo Fix the dumper, the output should not be ''.
+        $expected = '';
+        $yaml = $this->dumper->dump($data, 2);
+        $this->assertSame($expected, $yaml);
```

Comment:
@stof Fyi I am following a philosophy where we also cover scenarios that currently don't work as intended, as a way to document known issues that we are not currently fixing.

The commit or PR that contains a fix or that changes the nature of the bug will have to include a change in this test, documenting the change of behavior.